src/components/POLine/Item/ItemForm.js

```diff
@@ -58,7 +92,61 @@ class ItemForm extends Component {
       }));
 
       dispatch(change('contributors', lineContributors));
+      this.setState(({ contributors: lineContributors }));
+    }
+    if (identifiers && identifiers.length) {
+      const lineidentifiers = identifiers.map(({ identifierTypeId, value }) => ({
+        productId: value,
+        productIdType: find(identifierTypes, { id: identifierTypeId }).value,
+      }));
+
+      dispatch(change('details.productIds', lineidentifiers));
+      this.setState(({ productIds: lineidentifiers }));
+    }
+  };
+
+  setInstanceId = () => {
+    const { dispatch, change, initialValues, stripes: { store } } = this.props;
+    const formValues = getFormValues('POLineForm')(store.getState());
+    const instanceId = this.state.instanceId || get(initialValues, 'instanceId', '');
+    const title = this.state.title || get(initialValues, 'title', '');
+    const publisher = this.state.publisher || get(initialValues, 'publisher', '');
+    const publicationDate = this.state.publicationDate || get(initialValues, 'publicationDate', '');
+    const edition = this.state.edition || get(initialValues, 'edition', '');
+    const contributors = this.state.contributors.length ? this.state.contributors : get(initialValues, 'contributors', []);
+    const productIds = this.state.productIds.length ? this.state.productIds : get(initialValues, 'details.productIds', []);
+    const isEqualContributors = contributors.every(el => {
+      const contributor = find(get(formValues, 'contributors', []), { 'contributor': el.contributor });
+
+      return contributor ? isEqual(contributor, el) : false;
+    });
+    const isEqualProductIds = productIds.every(el => {
+      const productId = find(get(formValues, 'details.productIds', []), { 'productId': el.productId });
+
+      return productId ? isEqual(productId, el) : false;
+    });
+
+    if (
+      instanceId
+      && (title === formValues.title)
+      && (publisher === formValues.publisher)
+      && (publicationDate === formValues.publicationDate)
+      && (edition === formValues.edition)
+      && isEqualContributors
+      && isEqualProductIds
+    ) {
+      dispatch(change('instanceId', instanceId));
+    } else {
+      dispatch(change('instanceId', ''));
     }
+  }
+
+  onChangeField = (value, fieldName) => {
+    const { dispatch, change } = this.props;
+
+    dispatch(change(fieldName, value));
```

Comment:
I'm not sure if this should be called

Reply:
I think it should be called, otherwise we'll get old values